### PR TITLE
Not export links to unexported children

### LINKED ISSUE #3218: Clean up mtpr in parent on export
Beim Export einer Überordnung (getriggert durch den Export einer Unterordnung, oder administrativ) muss geprüft werden, welche Unterordnungen bereits exportiert wurden (exported-flag), diese metspointer müssen in der metsdatei erhalten bleiben. die metspointer nicht exportierter Vorgänge müssen beim export der überordnung aus der mets-datei entfernt werden.

### PR DESCRIPTION
**Pull request by matthias-ronge**
Resolves #3218

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/production/services/schema/SchemaService.java [hunk 1]
```diff
@@ -225,18 +226,24 @@ private void addUse(Subfolder useFolder, String canonical, MediaUnit mediaUnit)
...
-    private void convertChildrenLinksForExport(Workpiece workpiece, IncludedStructuralElement structure,
+    private boolean convertChildrenLinksForExport(Workpiece workpiece, IncludedStructuralElement structure,
                                                LegacyPrefsHelper prefs) throws DAOException, IOException {
 
         LinkedMetsResource link = structure.getLink();
         if (Objects.nonNull(link)) {
             int linkedProcessId = ServiceManager.getProcessService().processIdFromUri(link.getUri());
             Process process = ServiceManager.getProcessService().getById(linkedProcessId);
+            if (!process.isExported()) {
+                return true;
```
[the remark this answers is not in the document]
Comment (matthias-ronge) [reply Kitodo/src/main/java/org/kitodo/production/services/schema/SchemaService.java]: This is to [avoid negative conditionals](https://blog.jetbrains.com/idea/2014/09/the-inspection-connection-issue-2/#crayon-5e68cf1e6ed6c761588991-11).

[the remark this answers is not in the document]
Comment (solth) [reply Kitodo/src/main/java/org/kitodo/production/services/schema/SchemaService.java]: `if (!process.isExported())` **is** a "negative conditional", isn't it?

[the remark this answers is not in the document]
Comment (matthias-ronge) [reply Kitodo/src/main/java/org/kitodo/production/services/schema/SchemaService.java]: Would you prefer a `process.isNotYetExported()`?

[the remark this answers is not in the document]
Comment (solth) [reply Kitodo/src/main/java/org/kitodo/production/services/schema/SchemaService.java]: What is wrong about 
```
if (process.isExported()) {
    return false;
}
```
?

File: Kitodo/src/main/java/org/kitodo/production/services/schema/SchemaService.java [hunk 2]
```diff
@@ -226,24 +226,38 @@ private void addUse(Subfolder useFolder, String canonical, MediaUnit mediaUnit)
...
             Process process = ServiceManager.getProcessService().getById(linkedProcessId);
             if (!process.isExported()) {
-                return true;
+                return false;
             }
             setLinkForExport(structure, process, prefs, workpiece);
         }
         for (Iterator<IncludedStructuralElement> iterator = structure.getChildren().iterator(); iterator.hasNext();) {
-            if (convertChildrenLinksForExport(workpiece, iterator.next(), prefs)) {
+            if (!convertChildrenLinksForExportRecursive(workpiece, iterator.next(), prefs)) {
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/services/schema/SchemaService.java]: now you have even two negative conditions. Please swap the if contents

File: Kitodo/src/main/java/org/kitodo/production/services/schema/SchemaService.java [hunk 3]
```diff
@@ -86,7 +86,7 @@
         set(workpiece, MdSec.TECH_MD, "purlUrl", vp.replace(process.getProject().getMetsPurl()));
         set(workpiece, MdSec.TECH_MD, "contentIDs", vp.replace(process.getProject().getMetsContentIDs()));
 
-        convertChildrenLinksForExport(workpiece, workpiece.getRootElement(), prefs);
+        convertChildrenLinksForExportRecursive(workpiece, workpiece.getRootElement(), prefs);
```
Comment (Kathrin-Huber) [inline Kitodo/src/main/java/org/kitodo/production/services/schema/SchemaService.java]: please don't use the word "recursive" in a method.

**Pull request merged**